de-lib/src/main/java/org/iplantc/de/preferences/client/view/PreferencesViewImpl.java

```diff
@@ -286,4 +387,19 @@ public void userSessionFail() {
     void onResetHpcToken(SelectEvent event) {
         fireEvent(new ResetHpcTokenClicked());
     }
+
+    @UiHandler("testBtn")
+    void onAddSlack(SelectEvent event) {
+        if (!Strings.isNullOrEmpty(hookUrl.getValue())) {
+            fireEvent(new TestWebhookClicked(hookUrl.getValue()));
+        } else {
+            hookUrl.markInvalid(appearance.validUrl());
```

Comment:
Doesn't the `UrlValidator` take care of marking the field?  I would think you'd call `hookUrl.isValid` and then fire an event or not based on that result.